src/EditorFeatures/CSharpTest/Diagnostics/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UseAutoProperty;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.UseAutoProperty
+{
+    public class UseAutoPropertyTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return Tuple.Create<DiagnosticAnalyzer, CodeFixProvider>(
+                new UseAutoPropertyAnalyzer(), new UseAutoPropertyCodeFixProvider());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestSingleGetter1()
+        {
+            Test(
+@"class Class { [|int i|]; int P { get { return i; } } }",
+@"class Class { int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestCSharp5()
+        {
+            TestMissing(
+@"class Class { [|int i|]; int P { get { return i; } } }",
+            CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp5));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestSingleGetter2()
+        {
+            Test(
+@"class Class { int i; [|int P { get { return i; } }|] }",
+@"class Class { int P { get; } }");
```

Comment:
Is this a test for spans from where the fix could be invoked? If I understood it right (from `TestSingleGetter1` and `TestSingleGetter2`) , the fix can be invoked from both the field and the property. If so, naming the tests as so would be nice. Certainly a nit though :)